Index on important terms (#185)
ietf-wg-ohai/oblivious-http: draft-ietf-ohai-ohttp.md

  Clients and a single Oblivious Gateway Resource.
  {: anchor="dfn-relay"}

Oblivious Gateway Resource:

: A resource that can receive an encapsulated request, extract the contents of
  that request, forward it to a Target Resource, receive a response,
  encapsulate that response, then return that response.
  {: anchor="dfn-gateway"}

Target Resource:

: The resource that is the target of an encapsulated request.  This resource
  logically handles only regular HTTP requests and responses and so might be
  ignorant of the use of Oblivious HTTP to reach it.
  {: anchor="dfn-target"}

Relay Request:

: An HTTP request from Client to Relay that contains an encapsulated request as the content.
  {: anchor="dfn-relay-req"}

Relay Response:

: An HTTP response from Relay to Client that contains an encapsulated response as the content.
  {: anchor="dfn-relay-res"}

Gateway Request:

: An HTTP request from Relay to Gateway that contains an encapsulated request as the content.
  {: anchor="dfn-gateway-req"}

Gateway Response:

: An HTTP response from Gateway to Relay that contains an encapsulated response as the content.

>   {: anchor="dfn-gateway-res"}

Author: @chris-wood How would you feel about deleting these Xxx Request/Response definitions.  The index shows that we don't use them at all.

Reviewer: If we don't use them then we can drop them. 👍

To achieve the stated privacy goals, the Oblivious Relay Resource cannot be
operated by the same entity as the Oblivious Gateway Resource. However,
colocation of the Oblivious Gateway Resource and Target Resource simplifies the
interactions between those resources without affecting Client privacy.

As a consequence of this configuration, Oblivious HTTP prevents linkability
described above. Informally, this means:

> 1. Requests and responses are known only to Clients and servers in possession

Reviewer: ```suggestion
1. Requests and responses are known only to Clients and Oblivious Gateway Resource in possession
```

Author: Isn't it the Target Resource also?  I left this because we defined "server" in this section as being an entity that operates both the gateway and the target.

Reviewer: It is, but that's true without OHTTP too, so I think it's only really useful to focus on what the new entities see here.

Author: Then maybe ...

```suggestion
1. Requests and responses are known only to Clients and Target Resources, plus Oblivious Gateway Resources that possess
```

>    of the corresponding response encapsulation key and HPKE keying material.

Reviewer: ```suggestion
   the corresponding response encapsulation key and HPKE keying material.
```

   In particular, the Oblivious Relay knows the origin and destination of an
   Encapsulated Request and Response, yet does not know the decrypted

>    contents. Likewise, targets know only the Oblivious Gateway origin, i.e.,

Reviewer: ```suggestion
   contents. Likewise, Oblivious Gateway Resources know only the Oblivious Relay Resource
```

>    the relay, and the decrypted request. Only the Client knows both the

Reviewer: ```suggestion
   and the decrypted request. Only the Client knows both the
```

>    contents. Likewise, Oblivious Gateway Resources know only the Oblivious Relay Resource

Author: ```suggestion
   contents. Likewise, Oblivious Gateway Resources learns only the Oblivious Relay Resource
```

>    plaintext request and response.

Reviewer: ```suggestion
   plaintext request and response and its identity.
```

Author: What is "its" in this context?  (Also, and and and is a bit strange.)

Reviewer: "it" is the client here. We say above that the gateway sees both plaintext request and response, but then here we say "only the client sees plaintext request and response," which is clearly a contradiction. (It's not _only_ the client.) So adding the fact that the client knows its own identity distinguishes its view from that of the gateway.

Author: Ok, maybe instead,

```suggestion
   plaintext request. No other entity that can see these plaintext messages can attribute them to the client.
```

Or drop the tweak. I worry that we might need to qualify this further: this somewhat depends on there being multiple concurrent users of the same relay making similarly sized requests. That sort of thing.

Reviewer: That's better, but still contradictory. The text still says that "_only_ the client knows both the plaintext request and response," which is false since both the gateway and target see the plaintext request and response. Maybe we just drop "Only the client..." and instead write "No entity that can see plaintext messages can attribute them to the client which generated them"?

Author: Sold.
